docs/blog/unettest.md

```diff
@@ -0,0 +1,288 @@
+want to support both the Monolith and the Microservice running simultaneously.
+As a DevOps engineer on staff, I was asked to write a rule in our NGINX
+configuration to mirror the requests--they should go to both the Microservice
+and the Monolith. This way, the Monolith would continue to work while the
+Microservice was under development.
+
+Seems easy enough. Referencing the NGINX docs, we could take the existing rule
+and add a `mirror` directive to a special mirror location
+
+```
+location ~* /api/v1/impotant_update/([^\/]*) {
+         proxy_pass $monolith$request_uri;
+
+         mirror /microservice_mirror/$1; # NEW LINE!
+
+         ...some other configurations
+}
+```
+
+and then define the mirror route like this
+
+```
+location ~* /microservice_mirror/(.*) {
+         internal;
+         proxy_pass $microservice/new/api/update?value=$1;
+
+         ...some other configurations
+}
+```
+
+> Note: I am trying to do a little funny business with the `$1`. The
+format of the new api is slightly different from the old one so I need to
+manipulate it. I know now (after making `unettest`) that this is the crux of my
+pain. It seems that you cannot pass `$1` to the _internal_ location block and
+parse it like you would with a regular route. It will be passed literally, as in
+`?value=$1` rather than expanding and interpolating the path part. I did not
+know what or why at this time.
+
+I fulfilled the requirements (or at least tried to):
+
++ send incoming request to two servers, (`monolith`) and (`microservice`)
++ slightly reconfigure the incoming request to fit the api difference
+
+With that done, I committed my NGINX changes and deployed to NonProd.  But... it
+did not work. The requests were not being mirrored correctly. The NGINX
+documentation, through complete and easy to reference, is anything but beginner
+friendly. I thought my conf files were not working due to my own ignorance. I
+tried to fix the conf files with that classic development strategy--guess and
+check.
+
+But as I guessed and checked, NGINX's behavior became stranger and stranger.
+
+It wasn't exactly seven-and-a-half million years to get the answers to my
+questions, but it sure felt like it.  My development process was not
+user-friendly and there was a high cost to check each guess. I had to:
+
++ Commit a change
++ Deploy to NonProd (~10 minutes total)
++ Run my manual tests
++ Examine CloudWatch logs to find the recorded behavior
+
+Searching CloudWatch is not an easy or reliable way to detect subtle nuances in
+configuration changes. Sometimes the test results would not appear at all,
+sometimes I would be searching old time frames, sometimes old tests would be
+mixed in with the current tests, I was flooded with unrelated logs. It was
+bad.
+
+It was a bad way of asking questions. This development cycle was so kludgy , I
+was able to suss out WHAT the behavior was but not WHY. Not even WHAT EXACTLY.
+
+## Life, the Universe, and Everything
+
```

Comment:
In this section, you revert back to the "I" voice. I think the change comes whenever you go from explaining a technical issue to telling a story. If you think about successful tech blog posts you've read, which of these two components is more compelling for you? Most likely the explanation of the technical issue. Personally, I enjoy reading your story, but for a wider audience, it's going to dilute what they're here for.